### PR TITLE
functors_task_1 - Obmok

### PR DESCRIPTION
**Pull request by soniaobmok**

### CODE REVIEW

File: functors_task_1/student/include/impl.h [hunk 1]
```diff
@@ -8,6 +8,18 @@
  */
 struct BeerOrganizer
 {
+    BeerBrand operator()()
+    {
+        ++brandIndex;
+        if (brandIndex == static_cast<size_t>(BeerBrand::Max))
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h:14]: please specify the possible range explicitly with "from" and "to" constexpr variables

File: functors_task_1/student/include/impl.h [hunk 2]
```diff
@@ -16,16 +28,22 @@ struct BeerOrganizer
...
+    }
+    return false;
 }
 
 /**
  * @todo Implement lambda beer country equality comparator
  * @return true if beer county is the same, false otherwise
  */
-auto sameCountry = [](/**???*/)
+auto sameCountry = [](const BeerBrand& lhs, const BeerBrand& rhs)
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h:44]: pass literals by copy:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const

File: functors_task_1/student/include/impl.h (hunk 1 above)
Comment (soniaobmok) [reply functors_task_1/student/include/impl.h]: done

File: functors_task_1/student/include/impl.h (hunk 2 above)
Comment (soniaobmok) [reply functors_task_1/student/include/impl.h]: done

File: functors_task_1/student/include/impl.h [hunk 3]
```diff
@@ -16,16 +31,22 @@ struct BeerOrganizer
  *
  * @note Only Corona and HoeGaarden are expensive
  */
-bool isExpensiveBeer(/**???*/)
+bool isExpensiveBeer(BeerBrand beer)
 {
+    if (beer == BeerBrand::Corona || beer == BeerBrand::HoeGaarden)
+    {
+        return true;
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h:38]: avoid explicit true/false in you code, just "return beer == Corona || ...."

File: functors_task_1/student/include/impl.h [hunk 4]
```diff
@@ -16,16 +31,22 @@ struct BeerOrganizer
  *
  * @note Only Corona and HoeGaarden are expensive
  */
-bool isExpensiveBeer(/**???*/)
+bool isExpensiveBeer(BeerBrand beer)
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h:34]: const

**Pull request closed without merging**